src/main/java/com/linkedin/kmf/services/ClusterTopicManipulationService.java

```diff
@@ -0,0 +1,242 @@
+          adminClient.describeLogDirs(Collections.singleton(broker.id())).all().get();
+
+      this.processBroker(logDirectoriesResponseMap, broker, topic);
+    }
+    return _totalPartitions.get() == 0;
+  }
+
+  void processBroker(Map<Integer, Map<String, DescribeLogDirsResponse.LogDirInfo>> logDirectoriesResponseMap,
+      Node broker, String topic) {
+    LOGGER.trace("logDirectoriesResponseMap: {}", logDirectoriesResponseMap);
+    Map<String, DescribeLogDirsResponse.LogDirInfo> logDirInfoMap = logDirectoriesResponseMap.get(broker.id());
+    String logDirectoriesKey = logDirInfoMap.keySet().iterator().next();
+    LOGGER.debug("logDirInfoMap: {}", logDirInfoMap.get(logDirectoriesKey));
+    DescribeLogDirsResponse.LogDirInfo logDirInfo = logDirInfoMap.get(logDirectoriesKey);
+
+    if (logDirInfo != null && !logDirectoriesResponseMap.isEmpty()) {
+      Map<TopicPartition, DescribeLogDirsResponse.ReplicaInfo> topicPartitionReplicaInfoMap = logDirInfo.replicaInfos;
+      this.processLogDirsWithinBroker(topicPartitionReplicaInfoMap, topic, broker);
+    }
+  }
+
+  private void processLogDirsWithinBroker(
+      Map<TopicPartition, DescribeLogDirsResponse.ReplicaInfo> topicPartitionReplicaInfoMap, String topic,
+      Node broker) {
+    for (Map.Entry<TopicPartition, DescribeLogDirsResponse.ReplicaInfo> topicPartitionReplicaInfoEntry : topicPartitionReplicaInfoMap
+        .entrySet()) {
+
+      TopicPartition topicPartition = topicPartitionReplicaInfoEntry.getKey();
+      DescribeLogDirsResponse.ReplicaInfo replicaInfo = topicPartitionReplicaInfoEntry.getValue();
+
+      if (topicPartition.topic().equals(topic)) {
+        _totalPartitions.getAndDecrement();
+        LOGGER.trace("_totalPartitions count = {}", _totalPartitions);
+      }
+
+      LOGGER.debug("broker information: {}", broker);
+      LOGGER.trace("logDirInfo for kafka-logs: topicPartition = {}, replicaInfo = {}", topicPartition, replicaInfo);
+    }
+  }
+
+  /**
+   * This may be called multiple times.  This method must be thread safe and must assume that start() may be called
+   * concurrently.  This can happen if the monitoring application's life cycle is being managed by a container.
+   * Implementations must be non-blocking and should release the resources acquired by the service during start().
+   */
+  @Override
+  public void stop() {
+    _executor.shutdown();
+  }
+
+  /**
+   * Implementations of this method must be thread safe as it can be called at any time.  Implementations must be
+   * non-blocking.
+   * @return true if this start() has returned successfully else this must return false.  This must also return false if
+   * the service can no longer perform its function.
+   */
+  @Override
+  public boolean isRunning() {
+
+    return !_executor.isShutdown();
+  }
+
+  /**
+   * Implementations of this method must be thread safe and must be blocking.
+   */
+  @Override
+  public void awaitShutdown() {
```

Comment:
I know this method is just implementing the interface method `public void awaitShutdown();` But I strongly encourage you to change the interface signature to let users pass in the timeout duration in the same way as the `awaitTermination(...)` on the executor interface.

Since `awaitShutdown()` without any timeout duration specified is just absurd to me and considering this repo is open-sourced, I really think we should make this change.